[Bug] another index out of bounds error

### Describe the bug

A facebook user sent the attached .ork to me.  Index out of bounds when simulating under current unstable. This design is really a torture test of our pod/booster code, so while it needs fixing I don't think it needs to be any sort of blocker.


net.sf.openrocket.gui.main.SwingExceptionHandler$InternalException: An exception occurred during the simulation
at net.sf.openrocket.gui.main.SwingExceptionHandler.handleErrorCondition(SwingExceptionHandler.java:120)
at net.sf.openrocket.gui.simulation.SimulationRunDialog$InteractiveSimulationWorker.simulationInterrupted(SimulationRunDialog.java:454)
at net.sf.openrocket.gui.simulation.SimulationWorker.done(SimulationWorker.java:109)
at java.desktop/javax.swing.SwingWorker$5.run(SwingWorker.java:750)
at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(SwingWorker.java:848)
at java.desktop/sun.swing.AccumulativeRunnable.run(AccumulativeRunnable.java:112)
at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(SwingWorker.java:858)
at java.desktop/javax.swing.Timer.fireActionPerformed(Timer.java:311)
at java.desktop/javax.swing.Timer$DoPostEvent.run(Timer.java:243)
at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:771)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:716)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:741)
at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:191)
at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:236)
at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:234)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:234)
at java.desktop/java.awt.Dialog.show(Dialog.java:1080)
at java.desktop/java.awt.Component.show(Component.java:1728)
at java.desktop/java.awt.Component.setVisible(Component.java:1675)
at java.desktop/java.awt.Window.setVisible(Window.java:1036)
at java.desktop/java.awt.Dialog.setVisible(Dialog.java:1016)
at net.sf.openrocket.gui.main.SimulationPanel.runSimulation(SimulationPanel.java:381)
at net.sf.openrocket.gui.main.SimulationPanel.access$900(SimulationPanel.java:85)
at net.sf.openrocket.gui.main.SimulationPanel$RunSimulationAction.actionPerformed(SimulationPanel.java:681)
at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972)
at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313)
at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279)
at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
at java.desktop/java.awt.Component.processMouseEvent(Component.java:6626)
at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3389)
at java.desktop/java.awt.Component.processEvent(Component.java:6391)
at java.desktop/java.awt.Container.processEvent(Container.java:2266)
at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001)
at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948)
at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575)
at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516)
at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780)
at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:716)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:746)
at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:744)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:743)
at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.lang.IndexOutOfBoundsException: Index 2 out of bounds for length 2
at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
at java.base/java.util.Objects.checkIndex(Objects.java:359)
at java.base/java.util.ArrayList.get(ArrayList.java:427)
at net.sf.openrocket.rocketcomponent.RocketComponent.getChild(RocketComponent.java:2030)
at net.sf.openrocket.rocketcomponent.AxialStage.getUpperStage(AxialStage.java:178)
at net.sf.openrocket.motor.IgnitionEvent$3.isActivationEvent(IgnitionEvent.java:42)
at net.sf.openrocket.simulation.MotorClusterState.testForIgnition(MotorClusterState.java:187)
at net.sf.openrocket.simulation.BasicEventSimulationEngine.handleEvents(BasicEventSimulationEngine.java:310)
at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulateLoop(BasicEventSimulationEngine.java:155)
at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulate(BasicEventSimulationEngine.java:110)
at net.sf.openrocket.document.Simulation.simulate(Simulation.java:408)
at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:55)
at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:24)
at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)

---------- Thread information ----------
Thread[AWT-EventQueue-0,6,main]

---------- System information ----------
OpenRocket version: 23.xx
OpenRocket source: default
OpenRocket location: /home/joseph/rockets/openrocket/git/openrocket/unstable/swing/build/jar/OpenRocket.jar
User-defined thrust curves location: /home/joseph/.openrocket/ThrustCurves
JOGL version: 2.5.0-rc-20230523
Current default locale: en_US
System properties:
file.encoding=UTF-8
file.separator=/
java.class.path=swing/build/jar/OpenRocket.jar
java.class.version=61.0
java.home=/usr/lib/jvm/java-17-openjdk-amd64
java.io.tmpdir=/tmp
java.library.path=/usr/local/lib:/usr/lib:/usr/java/packages/lib:/usr/lib/x86_64-linux-gnu/jni:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/usr/lib/jni:/lib:/usr/lib
java.runtime.name=OpenJDK Runtime Environment
java.runtime.version=17.0.6+10-Debian-1
java.specification.name=Java Platform API Specification
java.specification.vendor=Oracle Corporation
java.specification.version=17
java.util.Arrays.useLegacyMergeSort=true
java.vendor=Debian
java.vendor.url=https://tracker.debian.org/openjdk-17
java.vendor.url.bug=https://bugs.debian.org/openjdk-17
java.version=17.0.6
java.version.date=2023-01-17
java.vm.compressedOopsMode=Zero based
java.vm.info=mixed mode, sharing
java.vm.name=OpenJDK 64-Bit Server VM
java.vm.specification.name=Java Virtual Machine Specification
java.vm.specification.vendor=Oracle Corporation
java.vm.specification.version=17
java.vm.vendor=Debian
java.vm.version=17.0.6+10-Debian-1
jdk.debug=release
jnlp.jogamp.tmp.cache.root=jln14223987386182612334
line.separator=\u000a
native.encoding=UTF-8
os.arch=amd64
os.name=Linux
os.version=6.1.0-8-amd64
path.separator=:
sun.arch.data.model=64
sun.awt.enableExtraMouseButtons=true
sun.awt.exception.handler=net.sf.openrocket.gui.main.SwingExceptionHandler$AwtHandler
sun.boot.library.path=/usr/lib/jvm/java-17-openjdk-amd64/lib
sun.cpu.endian=little
sun.font.fontmanager=sun.awt.X11FontManager
sun.io.unicode.encoding=UnicodeLittle
sun.java.command=swing/build/jar/OpenRocket.jar
sun.java.launcher=SUN_STANDARD
sun.jnu.encoding=UTF-8
sun.management.compiler=HotSpot 64-Bit Tiered Compilers
sun.stderr.encoding=UTF-8
sun.stdout.encoding=UTF-8
user.country=US
user.dir=/home/joseph/rockets/openrocket/git/openrocket/unstable
user.home=/home/joseph
user.language=en
user.name=joseph
user.timezone=America/Denver
---------- Error log ----------
1 0.001 INFO (SwingStartup.java:67) Starting up OpenRocket version 23.xx
2 0.003 INFO (SwingStartup.java:76) Checking for graphics head
3 0.004 INFO (SwingStartup.java:87) Moving startup to EDT
4 0.080 INFO (SwingStartup.java:179) Initializing the splash screen
5 0.138 INFO (SwingStartup.java:183) Registering exception handler
6 0.142 INFO (SwingStartup.java:189) Loading databases
7 1.026 INFO (TranslatorProvider.java:49) Using default locale en_US
8 1.037 INFO (TranslatorProvider.java:61) Set up translation for locale en_US, debug.currentFile=messages.properties
9 1.038 INFO (ComponentPresetDatabaseLoader.java:94) Loading component presets from datafiles/components
10 1.038 INFO (MotorDatabaseLoader.java:90) Starting reading serialized motor database
11 1.039 INFO (SwingStartup.java:251) Starting update check
17 1.041 INFO (SwingStartup.java:202) Setting best LAF
18 1.142 STDERR (GUIUtil.java:290) Error parsing gtk-icon-sizes string: ''
19 1.230 INFO (MotorDatabaseLoader.java:96) Ending reading serialized motor database, motorCount=1896
20 1.233 INFO (MotorDatabaseLoader.java:71) Starting reading user-defined motors
24 1.256 INFO (MotorDatabaseLoader.java:81) Ending reading user-defined motors, motorCount=1903
100 1.547 INFO (SwingStartup.java:219) Opening main application window
101 1.559 INFO (BasicFrame.java:1795) New action initiated
137 2.330 WARN (UpdateInfoRetriever.java:450) Malformed release tag: '23.xx'
188 4.789 INFO (SwingStartup.java:227) Checking update status
189 4.793 INFO (SwingStartup.java:95) Startup complete
193 16.055 INFO (BasicFrame.java:1282) Starting OpenFileWorker
194 16.056 INFO (SwingWorkerDialog.java:133) Running SwingWorker net.sf.openrocket.gui.util.OpenFileWorker@5f94029e
195 16.059 INFO (OpenRocketLoader.java:44) Loading .ork file
244 0.000 DEBUG (-) ===== 155 DEBUG lines removed =====
244 16.184 DEBUG (DatabaseMotorFinder.java:47) type Single-use, manufacturer Estes, designation E12, diameter NaN, length NaN, digest 27575268e95ac6983801956efb389764, warnings Messages[]
245 16.184 DEBUG (DatabaseMotorFinder.java:67) motor is E12
246 16.190 DEBUG (DatabaseMotorFinder.java:47) type Single-use, manufacturer Estes, designation C5, diameter NaN, length NaN, digest 73a9ed8bd883e2de7c1684f76fb54ab3, warnings Messages[]
247 16.191 DEBUG (DatabaseMotorFinder.java:47) type Single-use, manufacturer Estes, designation C6, diameter NaN, length NaN, digest c8743ef3fa99e14a89885cbe0ead47b2, warnings Messages[]
248 16.192 DEBUG (DatabaseMotorFinder.java:47) type Single-use, manufacturer Estes, designation C5, diameter NaN, length NaN, digest 73a9ed8bd883e2de7c1684f76fb54ab3, warnings Messages[]
249 16.192 DEBUG (DatabaseMotorFinder.java:47) type Single-use, manufacturer Estes, designation C6, diameter NaN, length NaN, digest c8743ef3fa99e14a89885cbe0ead47b2, warnings Messages[]
250 16.222 INFO (OpenRocketLoader.java:96) Loading done
251 16.256 INFO (SwingWorkerDialog.java:151) Worker completed before opening dialog
252 16.256 INFO (BasicFrame.java:1323) Warnings while reading file: Messages[No matching ComponentPreset for component Parachute kit, plastic, 24 in., PN 85568 found matching Estes PK-24A, 85568]
253 26.084 DEBUG (BasicFrame.java:1336) Opening new frame with the document
254 26.085 DEBUG (BasicFrame.java:180) Instantiating new BasicFrame
255 26.085 DEBUG (BasicFrame.java:191) Constructing the BasicFrame UI
271 26.145 DEBUG (RocketFigure3d.java:141) Initializing RocketFigure3D OpenGL Canvas
272 26.145 DEBUG (RocketFigure3d.java:143) Setting up GL capabilities...
300 26.219 INFO (RocketPanel.java:973) using pre-existing simulation
305 26.226 DEBUG (JarUtil.java:40) Found jar file using codeSource
306 26.227 DEBUG (ExampleDesignFile.java:118) Current jar file is: /home/joseph/rockets/openrocket/git/openrocket/unstable/swing/build/jar/OpenRocket.jar
314 26.264 DEBUG (BasicFrame.java:312) ... Setting Initial Selection: [Rocket, Sustainer]
315 26.264 DEBUG (BasicFrame.java:316) BasicFrame instantiation complete
316 26.268 INFO (BasicFrame.java:1204) Closing window because it is replaceable
317 26.268 DEBUG (BasicFrame.java:1732) Disposing window
324 26.525 DEBUG (Simulation.java:406) Simulation: calling simulator
325 26.533 INFO (BasicEventSimulationEngine.java:108) >>Starting simulation of branch: Sustainer
329 26.540 INFO (BasicEventSimulationEngine.java:325) Queueing Ignition Event for: 18mm Cluster Booster/ecb282a9 / C6 - Armed @: 0.0
330 26.540 INFO (BasicEventSimulationEngine.java:325) Queueing Ignition Event for: 24mm Booster/6c02f180 / E12 - Armed @: 0.0
335 26.541 INFO (BasicEventSimulationEngine.java:400) Igniting motor: 18mm Cluster Booster/ecb282a9 / C6 - Armed @0.0
339 26.541 INFO (BasicEventSimulationEngine.java:400) Igniting motor: 24mm Booster/6c02f180 / E12 - Armed @0.0
343 26.549 DEBUG (RK4SimulationStepper.java:187) Thrust estimate differs from correct value by 81.4%, estimate=5.072559785624213 correct=0.9428380893300248 timestep=0.01, recomputing k1 parameters
945 26.738 DEBUG (BasicEventSimulationEngine.java:449) detected Motor Burnout for motor C6@ 1.86 on stage 3: Booster
1041 26.759 DEBUG (BasicEventSimulationEngine.java:449) detected Motor Burnout for motor E12@ 2.44 on stage 3: Booster
1047 33.997 DEBUG (TextureCache.java:119) Loading texture Texture [offset=(0.00000,0.00000,0.00000), center=(0.00000,0.00000,0.00000), scale=(1.00000,1.00000,0.00000), rotation=0.0, image=/datafiles/textures/motors/estes.jpg]
1048 34.022 DEBUG (TextureCache.java:119) Loading texture Texture [offset=(0.00000,0.00000,0.00000), center=(0.00000,0.00000,0.00000), scale=(1.00000,1.00000,0.00000), rotation=0.0, image=/datafiles/textures/spiral-wound-alpha.png]
1049 34.046 DEBUG (TextureCache.java:119) Loading texture Texture [offset=(0.00000,0.00000,0.00000), center=(0.00000,0.00000,0.00000), scale=(1.00000,1.00000,0.00000), rotation=0.0, image=/datafiles/textures/wadding.png]
1050 34.057 DEBUG (TextureCache.java:119) Loading texture Texture [offset=(0.00000,0.00000,0.00000), center=(0.00000,0.00000,0.00000), scale=(1.00000,1.00000,0.00000), rotation=0.0, image=/datafiles/textures/balsa.jpg]
1051 34.085 DEBUG (TextureCache.java:119) Loading texture Texture [offset=(0.00000,0.00000,0.00000), center=(0.00000,0.00000,0.00000), scale=(1.00000,1.00000,0.00000), rotation=0.0, image=/datafiles/textures/chute.jpg]
1052 34.099 DEBUG (TextureCache.java:119) Loading texture Texture [offset=(0.00000,0.00000,0.00000), center=(0.00000,0.00000,0.00000), scale=(1.00000,1.00000,0.00000), rotation=0.0, image=/datafiles/textures/hardboard.jpg]
1053 34.110 DEBUG (TextureCache.java:119) Loading texture Texture [offset=(0.00000,0.00000,0.00000), center=(0.00000,0.00000,0.00000), scale=(1.00000,1.00000,0.00000), rotation=0.0, image=/datafiles/textures/wood.jpg]
1054 34.130 DEBUG (RocketFigure3d.java:411) Redrawing Overlay
1057 37.047 DEBUG (TextureCache.java:51) ClearCaches
1058 37.047 DEBUG (TextureCache.java:51) ClearCaches
1059 37.047 DEBUG (TextureCache.java:53) Destroying Texture for /datafiles/textures/spiral-wound-alpha.png
1060 37.047 DEBUG (TextureCache.java:53) Destroying Texture for /datafiles/textures/hardboard.jpg
1061 37.047 DEBUG (TextureCache.java:53) Destroying Texture for /datafiles/textures/wood.jpg
1062 37.047 DEBUG (TextureCache.java:53) Destroying Texture for /datafiles/textures/motors/estes.jpg
1063 37.047 DEBUG (TextureCache.java:53) Destroying Texture for /datafiles/textures/chute.jpg
1064 37.048 DEBUG (TextureCache.java:53) Destroying Texture for /datafiles/textures/wadding.png
1065 37.048 DEBUG (TextureCache.java:53) Destroying Texture for /datafiles/textures/balsa.jpg
1066 37.048 DEBUG (TextureCache.java:51) ClearCaches
1067 37.048 DEBUG (TextureCache.java:51) ClearCaches
1070 43.634 DEBUG (Simulation.java:406) Simulation: calling simulator
1071 43.636 INFO (BasicEventSimulationEngine.java:108) >>Starting simulation of branch: Sustainer
1075 43.638 INFO (BasicEventSimulationEngine.java:325) Queueing Ignition Event for: 24mm Booster/6c02f180 / D12 - Armed @: 0.0
1076 43.638 INFO (BasicEventSimulationEngine.java:325) Queueing Ignition Event for: 18mm Cluster Booster/ecb282a9 / C5 - Armed @: 0.0
1077 43.639 DEBUG (SimulationRunDialog.java:487) LAUNCH
1082 43.639 INFO (BasicEventSimulationEngine.java:400) Igniting motor: 24mm Booster/6c02f180 / D12 - Armed @0.0
1086 43.639 INFO (BasicEventSimulationEngine.java:400) Igniting motor: 18mm Cluster Booster/ecb282a9 / C5 - Armed @0.0
1088 43.640 DEBUG (SimulationRunDialog.java:242) Progressbar value 0
1089 43.640 DEBUG (SimulationRunDialog.java:247) No simulation status data available, setting empty labels
1092 43.640 DEBUG (RK4SimulationStepper.java:187) Thrust estimate differs from correct value by 58.4%, estimate=8.758397425373134 correct=3.645248626373626 timestep=0.01, recomputing k1 parameters
1594 0.000 VBOSE (-) ===== 1355 VBOSE lines removed =====
1594 43.743 VBOSE (RK4SimulationStepper.java:170) Selected time step 0.033900918279380216 (limiting factor 5)
1595 43.743 VBOSE (RK4SimulationStepper.java:182) Thrust = 16.544035076200007
1596 43.744 VBOSE (BasicEventSimulationEngine.java:297) HandleEvents: current branch = Sustainer
1597 43.744 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=ALTITUDE,time=1.3789395392027863,source=Rocket,data=[57.462580552505756;59.561767418445726]]
1598 43.744 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = [FlightEvent[type=BURNOUT,time=1.65,source=24mm Booster,data=D12], FlightEvent[type=RECOVERY_DEVICE_DEPLOYMENT,time=4.5,source=Parachute kit, plastic, 24 in., PN 85568,data=null], FlightEvent[type=BURNOUT,time=1.992,source=18mm Cluster Booster,data=C5]]
1599 43.744 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=ALTITUDE,time=1.3789395392027863,source=Rocket,data=[57.462580552505756;59.561767418445726]]
1600 43.744 VBOSE (BasicEventSimulationEngine.java:170) Taking simulation step at t=1.3789395392027863 altitude 59.561767418445726
1601 43.744 VBOSE (RK4SimulationStepper.java:170) Selected time step 0.05 (limiting factor 0)
1602 43.744 VBOSE (RK4SimulationStepper.java:182) Thrust = 16.43539802785162
1603 43.746 VBOSE (BasicEventSimulationEngine.java:297) HandleEvents: current branch = Sustainer
1604 43.746 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=ALTITUDE,time=1.4289395392027864,source=Rocket,data=[59.561767418445726;62.68465647008993]]
1605 43.746 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = [FlightEvent[type=BURNOUT,time=1.65,source=24mm Booster,data=D12], FlightEvent[type=RECOVERY_DEVICE_DEPLOYMENT,time=4.5,source=Parachute kit, plastic, 24 in., PN 85568,data=null], FlightEvent[type=BURNOUT,time=1.992,source=18mm Cluster Booster,data=C5]]
1606 43.746 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=ALTITUDE,time=1.4289395392027864,source=Rocket,data=[59.561767418445726;62.68465647008993]]
1607 43.746 VBOSE (BasicEventSimulationEngine.java:170) Taking simulation step at t=1.4289395392027864 altitude 62.68465647008993
1608 43.746 VBOSE (RK4SimulationStepper.java:170) Selected time step 0.05 (limiting factor 0)
1609 43.746 VBOSE (RK4SimulationStepper.java:182) Thrust = 16.34396609232482
1610 43.747 VBOSE (BasicEventSimulationEngine.java:297) HandleEvents: current branch = Sustainer
1611 43.747 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=ALTITUDE,time=1.4789395392027864,source=Rocket,data=[62.68465647008993;65.8375441505531]]
1612 43.747 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = [FlightEvent[type=BURNOUT,time=1.65,source=24mm Booster,data=D12], FlightEvent[type=RECOVERY_DEVICE_DEPLOYMENT,time=4.5,source=Parachute kit, plastic, 24 in., PN 85568,data=null], FlightEvent[type=BURNOUT,time=1.992,source=18mm Cluster Booster,data=C5]]
1613 43.747 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=ALTITUDE,time=1.4789395392027864,source=Rocket,data=[62.68465647008993;65.8375441505531]]
1614 43.747 VBOSE (BasicEventSimulationEngine.java:170) Taking simulation step at t=1.4789395392027864 altitude 65.8375441505531
1615 43.747 VBOSE (RK4SimulationStepper.java:170) Selected time step 0.04043415143735369 (limiting factor 5)
1616 43.747 VBOSE (RK4SimulationStepper.java:182) Thrust = 16.3191554439026
1617 43.748 VBOSE (BasicEventSimulationEngine.java:297) HandleEvents: current branch = Sustainer
1618 43.748 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=ALTITUDE,time=1.51937369064014,source=Rocket,data=[65.8375441505531;68.40799471842068]]
1619 43.748 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = [FlightEvent[type=BURNOUT,time=1.65,source=24mm Booster,data=D12], FlightEvent[type=RECOVERY_DEVICE_DEPLOYMENT,time=4.5,source=Parachute kit, plastic, 24 in., PN 85568,data=null], FlightEvent[type=BURNOUT,time=1.992,source=18mm Cluster Booster,data=C5]]
1620 43.748 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=ALTITUDE,time=1.51937369064014,source=Rocket,data=[65.8375441505531;68.40799471842068]]
1621 43.748 VBOSE (BasicEventSimulationEngine.java:170) Taking simulation step at t=1.51937369064014 altitude 68.40799471842068
1622 43.748 VBOSE (RK4SimulationStepper.java:170) Selected time step 0.05 (limiting factor 0)
1623 43.748 VBOSE (RK4SimulationStepper.java:182) Thrust = 16.295373552959948
1624 43.749 VBOSE (BasicEventSimulationEngine.java:297) HandleEvents: current branch = Sustainer
1625 43.749 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=ALTITUDE,time=1.56937369064014,source=Rocket,data=[68.40799471842068;71.61118635395054]]
1626 43.749 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = [FlightEvent[type=BURNOUT,time=1.65,source=24mm Booster,data=D12], FlightEvent[type=RECOVERY_DEVICE_DEPLOYMENT,time=4.5,source=Parachute kit, plastic, 24 in., PN 85568,data=null], FlightEvent[type=BURNOUT,time=1.992,source=18mm Cluster Booster,data=C5]]
1627 43.749 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=ALTITUDE,time=1.56937369064014,source=Rocket,data=[68.40799471842068;71.61118635395054]]
1628 43.749 VBOSE (BasicEventSimulationEngine.java:170) Taking simulation step at t=1.56937369064014 altitude 71.61118635395054
1629 43.749 VBOSE (RK4SimulationStepper.java:170) Selected time step 0.05 (limiting factor 0)
1630 43.749 VBOSE (RK4SimulationStepper.java:182) Thrust = 14.8558127591748
1631 43.750 VBOSE (BasicEventSimulationEngine.java:297) HandleEvents: current branch = Sustainer
1632 43.750 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=ALTITUDE,time=1.6193736906401401,source=Rocket,data=[71.61118635395054;74.83497125831126]]
1633 43.750 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = [FlightEvent[type=BURNOUT,time=1.65,source=24mm Booster,data=D12], FlightEvent[type=RECOVERY_DEVICE_DEPLOYMENT,time=4.5,source=Parachute kit, plastic, 24 in., PN 85568,data=null], FlightEvent[type=BURNOUT,time=1.992,source=18mm Cluster Booster,data=C5]]
1634 43.750 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=ALTITUDE,time=1.6193736906401401,source=Rocket,data=[71.61118635395054;74.83497125831126]]
1635 43.750 VBOSE (BasicEventSimulationEngine.java:170) Taking simulation step at t=1.6193736906401401 altitude 74.83497125831126
1636 43.750 VBOSE (RK4SimulationStepper.java:170) Selected time step 0.03062630935985977 (limiting factor 1)
1637 43.750 VBOSE (RK4SimulationStepper.java:182) Thrust = 9.853713116421773
1638 43.751 VBOSE (BasicEventSimulationEngine.java:297) HandleEvents: current branch = Sustainer
1639 43.751 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=BURNOUT,time=1.65,source=24mm Booster,data=D12]
1640 43.751 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = [FlightEvent[type=ALTITUDE,time=1.65,source=Rocket,data=[74.83497125831126;76.80934886645798]], FlightEvent[type=RECOVERY_DEVICE_DEPLOYMENT,time=4.5,source=Parachute kit, plastic, 24 in., PN 85568,data=null], FlightEvent[type=BURNOUT,time=1.992,source=18mm Cluster Booster,data=C5]]
1641 43.751 VBOSE (BasicEventSimulationEngine.java:383) Handling event FlightEvent[type=BURNOUT,time=1.65,source=24mm Booster,data=D12]
1642 43.751 DEBUG (BasicEventSimulationEngine.java:449) detected Motor Burnout for motor D12@ 1.65 on stage 3: Booster
1643 43.751 VBOSE (BasicEventSimulationEngine.java:299) Obtained event from queue: FlightEvent[type=EJECTION_CHARGE,time=1.65,source=Booster,data=D12]
1644 43.751 VBOSE (BasicEventSimulationEngine.java:300) Remaining EventQueue = [FlightEvent[type=ALTITUDE,time=1.65,source=Rocket,data=[74.83497125831126;76.80934886645798]], FlightEvent[type=RECOVERY_DEVICE_DEPLOYMENT,time=4.5,source=Parachute kit, plastic, 24 in., PN 85568,data=null], FlightEvent[type=BURNOUT,time=1.992,source=18mm Cluster Booster,data=C5]]
1645 43.805 DEBUG (SimulationRunDialog.java:354) updating simulationMaxAltitude[0] to 9.135421326597253E-6
1646 43.805 DEBUG (SimulationRunDialog.java:366) simulationStage boost: t=0.01 est=6.514370958195171
1647 43.805 DEBUG (SimulationRunDialog.java:464) Setting progress to 0 (real 0)
1648 43.805 DEBUG (SimulationRunDialog.java:242) Progressbar value 0
1649 43.807 ERROR (SwingExceptionHandler.java:119) An exception occurred during the simulation
> java.lang.IndexOutOfBoundsException: Index 2 out of bounds for length 2
> at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
> at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
> at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
> at java.base/java.util.Objects.checkIndex(Objects.java:359)
> at java.base/java.util.ArrayList.get(ArrayList.java:427)
> at net.sf.openrocket.rocketcomponent.RocketComponent.getChild(RocketComponent.java:2030)
> at net.sf.openrocket.rocketcomponent.AxialStage.getUpperStage(AxialStage.java:178)
> at net.sf.openrocket.motor.IgnitionEvent$3.isActivationEvent(IgnitionEvent.java:42)
> at net.sf.openrocket.simulation.MotorClusterState.testForIgnition(MotorClusterState.java:187)
> at net.sf.openrocket.simulation.BasicEventSimulationEngine.handleEvents(BasicEventSimulationEngine.java:310)
> at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulateLoop(BasicEventSimulationEngine.java:155)
> at net.sf.openrocket.simulation.BasicEventSimulationEngine.simulate(BasicEventSimulationEngine.java:110)
> at net.sf.openrocket.document.Simulation.simulate(Simulation.java:408)
> at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:55)
> at net.sf.openrocket.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:24)
> at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
> at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
> at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
> at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
> at java.base/java.lang.Thread.run(Thread.java:833)
1650 43.807 INFO (SwingExceptionHandler.java:142) Running in EDT, showing dialog
1651 43.807 INFO (SwingExceptionHandler.java:201) Showing Exception dialog
1652 48.114 USER (SwingExceptionHandler.java:220) User requested sending bug report
1653 48.116 DEBUG (JarUtil.java:40) Found jar file using codeSource
---------- End of bug report ----------
[Radical-Rocketry-Starstreak-BT-80-BT-5-rev-boosters.zip](https://github.com/openrocket/openrocket/files/12140487/Radical-Rocketry-Starstreak-BT-80-BT-5-rev-boosters.zip)


### To Reproduce

1. 
2. 
3. 
...


### Screenshots / .ork file

#### Screenshot(s):
*(drag-and-drop the screenshot(s) here)*


#### .ork file:
*(drag-and-drop the file here as a .zip file)*


### OpenRocket version

unstable

### What platform are you running on?

Linux

### Additional context

_No response_

--- a/core/src/net/sf/openrocket/rocketcomponent/AxialStage.java
+++ b/core/src/net/sf/openrocket/rocketcomponent/AxialStage.java
@@ -170,14 +170,14 @@ public class AxialStage extends ComponentAssembly implements FlightConfigurableC
 	 * @return	the previous stage in the rocket
 	 */
 	public AxialStage getUpperStage() {
-		if( null == this.parent ) {
+		if (this.parent == null) {
 			return null; 
-		}else if(Rocket.class.isAssignableFrom(this.parent.getClass()) ){
-			final int thisIndex = getStageNumber();
-			if( 0 < thisIndex ){
-				return (AxialStage)parent.getChild(thisIndex-1);
+		} else if (Rocket.class.isAssignableFrom(this.parent.getClass())) {
+			final int thisIndex = parent.getChildPosition(this);
+			if (thisIndex > 0) {
+				return (AxialStage) parent.getChild(thisIndex-1);
 			}
-		}else {
+		} else {
 			return this.parent.getStage();
 		}
 		return null;

